pkg/server/backuprestoreserver.go

```diff
@@ -275,17 +275,26 @@ func (b *BackupRestoreServer) runEtcdProbeLoopWithSnapshotter(ctx context.Contex
 			}
 		}
 
-		// set server's healthz endpoint status to OK so that
-		// etcd is marked as ready to serve traffic
-		handler.SetStatus(http.StatusOK)
-
 		if runServerWithSnapshotter {
 			if b.ownerChecker != nil {
 				// Check if the actual owner ID matches the expected one
 				// If the check fails or returns false, take a final full snapshot if needed
 				b.logger.Debugf("Checking owner before starting snapshotter...")
 				result, err := b.ownerChecker.Check(ctx)
-				if err != nil || !result {
+				if err != nil {
+					b.logger.Errorf("ownerChecker check fails: %v", err)
+
+					// Wait for the configured interval before making another attempt
+					b.logger.Infof("Waiting for %s...", b.config.OwnerCheckConfig.OwnerCheckInterval.Duration)
+					select {
+					case <-ctx.Done():
+						b.logger.Info("Shutting down...")
+						return
+					case <-time.After(b.config.OwnerCheckConfig.OwnerCheckInterval.Duration):
+					}
+					continue
```

Comment:
Why do we need this retry logic here? If the etcd gets restarted while there are connectivity problems to the DNS, it will not be able to start again.

Reply:
why etcd will not be able to start again?

Reply:
I meant the following scenario:
1. There are some errors with DNS on the landscape. 
2. For some other reason etcd-backup-restore is restarted
3. When etcd-backup-restore starts it will try to do the owner check on line 283
4. If the owner check returns an error, the select/case will block until the `b.config.OwnerCheckConfig.OwnerCheckInterval.Duration` has expired
5. Once the time expires, the loop is repeated, if there is yet again an error from the owner check then the select/case will block the loop again. 
6. Steps 4 and 5 will be repeated until the DNS errors are resolved and therefore `handler.SetStatus(http.StatusOK)` is not called and etcd-backup-restore does not become ready and snapshotter is not started

Reply:
make sense.